presto-tests/src/test/java/io/prestosql/tests/TestMetadataManager.java

```diff
@@ -151,4 +152,15 @@ public void testUpperCaseSchemaIsChangedToLowerCase()
                             return null;
                         });
     }
+
+    @Test
+    public void testUpperCaseListTablesFilter()
+    {
+        assertThat(queryRunner.execute("SELECT * FROM system.jdbc.tables WHERE TABLE_SCHEM = 'upper_case_schema' AND TABLE_NAME = 'upper_case_table'"))
+                .hasSize(1);
+        assertThat(queryRunner.execute("SELECT * FROM system.jdbc.tables WHERE TABLE_SCHEM = 'UPPER_CASE_SCHEMA'"))
```

Comment:
Once #17 got fixed it should return actually tables from `UPPER_CASE_SCHEMA`. Please add a comment and link to the #17 issue.